Functions/Quantification/xASL_quant_FSL.m

```diff
@@ -70,13 +90,31 @@
     % Remove residual BASIL-related files
     xASL_delete(pathFSLOptions);
     xASL_delete(pathFSLInput);
-	xASL_delete(pathFSLOutput);
+	xASL_delete(pathFSLOutput, 1);
     %xASL_adm_DeleteFileList(x.dir.SESSIONDIR, '(?i)^.*basil.*$', 1, [0 Inf]);
     
-    %% 3. Write the PWI as Nifti file for Basil/Fabber to read as input
+
+    %% 3. Write the PWI4D as Nifti file for Basil/Fabber to read as input
+    PWI4D_nii = xASL_io_ReadNifti(path_PWI4D);
+    PWI4D = PWI4D_nii.dat(:,:,:,:,:,:);
+    
     % FIXME would be good to have a brain mask at this point -> PM: if this would be a brainmask as well, we can skip creating a dummy input image here
     
-    PWI(isnan(PWI)) = 0;
+    % First, we extrapolate values to fill NaNs with a small kernel only, inside the brainmask
+    % We don't have a brainmask here yet, so now we just run this small kernel once
+    voxelSize = PWI4D_nii.hdr.pixdim(2:4);
+    kernelSize = round([8 8 8]./voxelSize);
+    for i4D = 1:size(PWI4D,4)
```

Comment:
I am not sure why was this added. This will not only fill NaNs with interpolated values, but it will also smooth the entire image.

This is strictly against our goals of keeping the images non-smoothed.
This has to be rewritten/removed

Reply:
Yes.  AFAIK this extrapolates + fills NaNs + smooths, that's why I complain about it. I wouldn't complain about extrapolation + filling.

Reply:
The ndnanfilter function has a setting that let's you choose between 1) smoothing only and ignoring NaNs 2) non-smoothing but extrapolating NaNs 3) both smoothing and extrapolating NaNs. AFAIK I selected number 2.

So are you asking me because you checked this and know, or am I know telling you something that you forgot (I'm pretty sure you used to know this parameter ;)

Reply:
Haha. Sorry, I forgot :) In that case - all is good :)

Reply:
%     WNAN   - Integer indicating NaNs treatment and program behaviour!:
%              0: Filters data and interpolates NaNs with values from non-NaNs (default). 
%              1: Filters data but leaves NaNs as NaNs
%              2: Leaves non-NaNs how they were and interpolate NaNs with values form non-NaNs
%              See the NOTEs below

we use option 2, so we're good. You can resolve this comment

Reply:
Yes, I saw. All good. I will resolve - just wanted you to see that I agree.